src/widget/wcoverartmenu.cpp

```diff
@@ -82,8 +86,38 @@ void WCoverArtMenu::slotChange() {
     coverInfo.source = CoverInfo::USER_SELECTED;
     coverInfo.coverLocation = selectedCoverPath;
     coverInfo.setImage(image);
-    qDebug() << "WCoverArtMenu::slotChange emit" << coverInfo;
-    emit coverInfoSelected(coverInfo);
+
+    QString coverArtCopyFilePath =
+            fileInfo.absoluteFilePath().left(
+                    fileInfo.absoluteFilePath().lastIndexOf('.') + 1) +
+            selectedCoverExtension;
+
+    if (QFileInfo(m_coverInfo.trackLocation).canonicalPath() ==
+            QFileInfo(selectedCoverPath).canonicalPath()) {
+        qDebug() << "Track and selected cover art are in the same path:"
+                 << QFileInfo(selectedCoverPath).canonicalPath()
+                 << "Cover art updated without copying";
+        emit coverInfoSelected(coverInfo);
+        qDebug() << "WCoverArtMenu::slotChange emit" << coverInfo;
```

Comment:
Is this a left over?

Reply:
Generally no.
If the question meant to be for the block. That was added if the cover art is the same path in the track, if that happens we will just update the cover art instead of copying it.

If the question meant to be for the debug message `slotChange emit` was placed related cover art updaters, as far as I tested this debug message always appears after each cover art update, so I didn't want to delete it.

Reply:
Oh yes, i see it above. OK.